src/NUnitFramework/framework/Constraints/NUnitEqualityComparer.cs

```diff
@@ -193,7 +205,7 @@ public bool AreEqual(object? x, object? y, ref Tolerance tolerance)
             static string GetType(object? x) => x?.GetType().FullName ?? "null";
         }
 
-        internal EqualMethodResult AreEqual(object? x, object? y, ref Tolerance tolerance, ComparisonState state)
+        internal EqualMethodResult AreEqual<T>(T? x, T? y, ref Tolerance tolerance, ComparisonState state)
```

Comment:
Oh right. I'd been thinking this would be no different than boxing at the assert level but I "think" what you're saying is that in some cases, like collections, this would be called O(n) times?

Reply:
Oh wait. I got it now